src/components/sidebar/WatchlistAddButton/index.tsx

```diff
@@ -1,34 +1,25 @@
+import { addSafeToWatchlist } from '@/components/new-safe/load/logic'
+import { useMnemonicSafeName } from '@/hooks/useMnemonicName'
+import useSafeInfo from '@/hooks/useSafeInfo'
 import { OVERVIEW_EVENTS, trackEvent } from '@/services/analytics'
-import { useRouter } from 'next/router'
-import { AppRoutes } from '@/config/routes'
-import { useCurrentChain } from '@/hooks/useChains'
-import useSafeAddress from '@/hooks/useSafeAddress'
 import { Button } from '@mui/material'
 import SafeListRemoveDialog from '../SafeListRemoveDialog'
-import { useAppSelector } from '@/store'
+import { useAppDispatch, useAppSelector } from '@/store'
 import { selectAddedSafes } from '@/store/addedSafesSlice'
 import { useState } from 'react'
 import { VisibilityOutlined } from '@mui/icons-material'
 
 const WatchlistAddButton = () => {
   const [open, setOpen] = useState(false)
-  const router = useRouter()
-  const chain = useCurrentChain()
-  const address = useSafeAddress()
-  const chainId = chain?.chainId || ''
-  const addedSafes = useAppSelector((state) => selectAddedSafes(state, chainId))
-  const isInWatchlist = !!addedSafes?.[address]
+  const { safe } = useSafeInfo()
+  const dispatch = useAppDispatch()
+  const safeName = useMnemonicSafeName()
```

Comment:
What if a Safe already has an actual name in the address book?

Reply:
Do you think its still useful to add the safe and its owners to the address book now that the "add safe" feature has become a watchlist i.e. is supposed to be used only for non-owned safes?

In any case lets check the address book for a name before defaulting to the mnemonic.